modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/storage/ConsensusStorage.scala

```diff
@@ -0,0 +1,24 @@
+package org.tessellation.dag.l1.domain.consensus.block.storage
+
+import cats.effect.{Ref, Sync}
+import cats.syntax.functor._
+import cats.syntax.option._
+
+import org.tessellation.dag.l1.domain.consensus.RoundId
+import org.tessellation.dag.l1.domain.consensus.block.RoundData
+
+import io.chrisdavenport.mapref.MapRef
+
+case class ConsensusStorage[F[_]](
+  ownConsensus: Ref[F, Option[RoundData]],
+  peerConsensuses: MapRef[F, RoundId, Option[RoundData]]
+)
```

Comment:
If it's a kind of service/storage/repository, I'd not use `case class`.

Reply:
I can use a `class` but both `ownConsensus` and `peerConsensuses` would still need to be `vals`, because I need to access them directly in `BlockConsensusCell`. Making proxy functions to access modify e.g. doesn't make sense I think.